pkg/stores/soughtgame/db.go

```diff
@@ -2,292 +2,353 @@ package soughtgame
+	if err != nil {
+		if err == pgx.ErrNoRows {
 			return nil, nil
 		}
 		return nil, err
 	}
 
-	receiverIsPermanent, err := rg.ReceiverIsPermanent()
+	sg.SeekRequest.ReceiverState = pb.SeekState_ABSENT
+	result, err := tx.Exec(ctx, `UPDATE soughtgames SET request = jsonb_set(request, array['receiver_state'], $1) WHERE receiver_conn_id = $2`, pb.SeekState_ABSENT, connID)
 	if err != nil {
 		return nil, err
 	}
+	if result.RowsAffected() != 1 {
+		return nil, fmt.Errorf("failed to update receiver status: %s (%d rows affected)", connID, result.RowsAffected())
+	}
 
-	rg.SeekRequest.ReceiverState = pb.SeekState_ABSENT
-	if !receiverIsPermanent {
-		rg.SeekRequest.ReceivingUser = nil
+	if err := tx.Commit(ctx); err != nil {
+		return nil, err
 	}
-	return rg, s.Set(ctx, rg)
+
+	return sg, nil
 }
 
 // ListOpenSeeks lists all open seek requests for receiverID, in tourneyID (optional)
 func (s *DBStore) ListOpenSeeks(ctx context.Context, receiverID, tourneyID string) ([]*entity.SoughtGame, error) {
-	var games []soughtgame
-	var err error
-	ctxDB := s.db.WithContext(ctx)
-	query := ctxDB.Table("soughtgames")
+	tx, err := s.dbPool.BeginTx(ctx, common.DefaultTxOptions)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback(ctx)
 
+	var rows pgx.Rows
 	if tourneyID != "" {
-		query = query.Where("receiver = ? AND request->>'tournament_id' = ?", receiverID, tourneyID)
+		rows, err = tx.Query(ctx, `SELECT request FROM soughtgames WHERE receiver = $1 AND request->>'tournament_id' = $2`, receiverID, tourneyID)
 	} else {
-		query = query.Where("receiver = ? OR receiver = ''", receiverID)
+		rows, err = tx.Query(ctx, `SELECT request FROM soughtgames WHERE receiver = $1 OR receiver = ''`, receiverID)
 	}
-	if result := query.Scan(&games); result.Error != nil {
-		return nil, result.Error
+	if err != nil {
+		return nil, err
 	}
+	defer rows.Close()
+
+	games := []*entity.SoughtGame{}
 
-	entGames := make([]*entity.SoughtGame, len(games))
-	for idx, g := range games {
-		entGames[idx], err = s.sgFromDBObj(&g)
-		if err != nil {
+	for rows.Next() {
+		var req pb.SeekRequest
+		if err := rows.Scan(&req); err != nil {
 			return nil, err
 		}
+		games = append(games, &entity.SoughtGame{SeekRequest: &req})
 	}
-	return entGames, nil
+	if err := tx.Commit(ctx); err != nil {
+		return nil, err
+	}
+	return games, nil
 }
 
 // ExistsForUser returns true if the user already has an outstanding seek request.
 func (s *DBStore) ExistsForUser(ctx context.Context, userID string) (bool, error) {
-	ctxDB := s.db.WithContext(ctx)
-	var count int64
-	if result := ctxDB.Model(&soughtgame{}).Where("seeker = ?", userID).Count(&count); result.Error != nil {
-		return false, result.Error
+	tx, err := s.dbPool.BeginTx(ctx, common.DefaultTxOptions)
+	if err != nil {
+		return false, err
 	}
-	return count > 0, nil
+	defer tx.Rollback(ctx)
+
+	var exists bool
+	err = tx.QueryRow(ctx, `SELECT EXISTS(SELECT 1 FROM soughtgames WHERE seeker = $1)`, userID).Scan(&exists)
```

Comment:
🤯

Reply:
I don't see it. What did I mess up here?

Reply:
No nothing, I just thought it was a cool query.

Reply:
:sunglasses: